File: Chapter10/ex02_practice10-2/src/main/java/com/example/ex02_practice10_2/ResultActivity.java
package com.example.ex02_practice10_2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.RatingBar;
import android.widget.TextView;


public class ResultActivity extends AppCompatActivity {

    Button btnReturn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result);

        btnReturn = findViewById(R.id.btnReturn);

        // MainActivity에서 보낸 인텐트 객체 받기
        Intent intent = getIntent();
        String[] imageNames = intent.getStringArrayExtra("nameData");
        int[] voteResult = intent.getIntArrayExtra("voteData");


        //region 뷰를 담을 참조변수 배열
        TextView[] textViews = new TextView[imageNames.length];
        RatingBar[] ratingBars = new RatingBar[voteResult.length];

        int[] textViewsId = {R.id.tv1,R.id.tv2,R.id.tv3,R.id.tv4,R.id.tv5,R.id.tv6,R.id.tv7,R.id.tv8,R.id.tv9};
        int[] ratingBarsId = {R.id.rbar1,R.id.rbar2,R.id.rbar3,R.id.rbar4,R.id.rbar5,R.id.rbar6,R.id.rbar7,R.id.rbar8,R.id.rbar9};

        for (int i=0; i<textViews.length; i++){
            textViews[i]=findViewById(textViewsId[i]);
            ratingBars[i]=findViewById(ratingBarsId[i]);
        }
        //endregion

        //TextView에는 이미지 이름, RatingBar에는 투표 결과를 출력
        for (int i=0; i<voteResult.length; i++){
            textViews[i].setText(imageNames[i]);
            ratingBars[i].setRating(voteResult[i]);
        }
        //돌아가기 버튼 이벤트 처리
        btnReturn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
    }
}

File: Practice_Exam/PracticeKotlin/app/src/main/java/com/example/practicekotlin/ResultActivity.kt
package com.example.practicekotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_result.*

class ResultActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_result)

        //버튼을 누르면 액티비티 변경해주기
        val bmi = intent.extras?.get("Bmi")
        when(bmi){
            0.0..18.0->{imageView.setImageResource(R.drawable.sad_face)}
            18.1..23.0 ->{imageView.setImageResource(R.drawable.smile_face)}
            else->{imageView.setImageResource(R.drawable.sad_face)}
        }

    }
}
